src/functions/function-command/build-and-deploy-function.ts

```diff
@@ -146,13 +149,29 @@ export async function buildFunction(context?: FunctionNode): Promise<void> {
   telemetryLog('function_build_command', 'Build command execute');
   // eslint-disable-next-line @typescript-eslint/no-floating-promises
   functionExplorer.refresh();
-  await knExecutor.executeInTerminal(
-    await FuncAPI.buildFunc(
-      context ? context.contextPath.fsPath : selectedFolderPick.workspaceFolder.uri.fsPath,
-      funcData.image,
-      funcData.builder,
-    ),
+  const command = await FuncAPI.buildFunc(
+    context ? context.contextPath.fsPath : selectedFolderPick.workspaceFolder.uri.fsPath,
+    funcData.image,
   );
+  const name = `Function ${command.cliArguments[0]}: ${context.getName()}`;
+  if (!STILL_EXECUTING_COMMAND.get(name)) {
+    // eslint-disable-next-line no-return-await
+    return await executeCommandInOutputChannels(command, name);
+  }
+  const status = await vscode.window.showWarningMessage(
+    `The Function ${command.cliArguments[0]}: ${context.getName()} is already active.`,
+    'Terminate',
+    'Restart',
+  );
+  if (status === 'Terminate') {
+    CACHED_CHILDPROCESS.get(name).kill('SIGTERM');
+  } else if (status === 'Restart') {
+    CACHED_CHILDPROCESS.get(name).kill('SIGTERM');
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    setTimeout(async () => {
```

Comment:
Wouldn't it be safer to listen to the end of `CACHED_CHILDPROCESS` instead of waiting 4sec to restart it?